addons/base_vat/models/res_partner.py

```diff
@@ -481,7 +481,7 @@ def create(self, values):
         return super(ResPartner, self).create(values)
 
     def write(self, values):
-        if values.get('vat'):
+        if values.get('vat') and (self.is_company or not self.parent_id):
```

Comment:
```suggestion
        if values.get('vat') and len(self.mapped('country_id')) == 1:
            country_id = values.get('country_id', self[:1].country_id.id)
```

Reply:
I agree with the workaround (I did not think about the programmatical way ^^)
I have adapted both the code and the commit message.